lib/parlour/types.rb

```diff
@@ -292,11 +329,11 @@ def ==(other)
     end
 
     # A hash with known key and value types.
-    class Hash < Type
+    class Hash < Generic
       sig { params(key: TypeLike, value: TypeLike).void }
       def initialize(key, value)
-        @key = to_type(key)
-        @value = to_type(value)
+        @key = key
+        @value = value
       end
```

Comment:
I think it's best to leave `Hash` as a subclass of `Type` rather than `Generic` for now, for consistency with `Array` and friends.
Also, I think this removal of `to_type` would cause problems if strings are passed in as raw types?

Reply:
Ah, looks like you've fixed this, ta :)